crates/red_knot_python_semantic/src/search.rs

```diff
@@ -0,0 +1,58 @@
+use crate::{
+    location::CanLocate,
+    semantic_index::{semantic_index, SemanticIndex},
+    types::Type,
+};
+use lsp_types::Position;
+use ruff_db::{
+    files::File,
+    parsed::parsed_module,
+    source::{line_index, source_text},
+};
+use ruff_python_ast::Identifier;
+use ruff_source_file::OneIndexed;
+
+use crate::{
+    location::{CPosition, DefLocation},
+    Db,
+};
+
+pub fn definition_at_location(file: File, location: Position, db: &dyn Db) -> Option<DefLocation> {
+    // XXX now this returns one or none. It could return an iterator of locations
```

Comment:
Not clear to me if our current system could multiple locations but the LSP can handle multiple definition sites (I imagine in case of ambiguity)